### ISSUE TITLE
[Enhancement]: Allow customization of the database container by supplying a customization function

### ISSUE THREAD

**Issue by bsofiato** [testcontainers/testcontainers-java#5737]
### Module

Core

### Proposal

We can customize the DB created by the Testcontainer's JDBC integration using a function passed on the database connection URL. 

What if we could customize the container using a similar scheme?

I propose we introduce a URL parameter named **TC_CONTAINERCUSTOMIZER** (maybe we can come up with a smaller name). It takes the fully qualified name of the function we want to call to customize the newly created container. For instance, using the URL below would call the `sampleCustomizer` method of the on `org.testcontainers.jdbc.JDBCDriverTest` before starting the database container.

`jdbc:tc:mysql:5.7.34:///databasename?TC_CONTAINERCUSTOMIZER =org.testcontainers.jdbc.JDBCDriverTest::sampleCustomizer `

The signature of the sampleCustomizer should be something like the snippet below.

```java
public class JDBCDriverTest {
    public <S extends JdbcDatabaseContainer<S>> static S sampleCustomizer (S container) throws Exception {
        // e.g. change parameters, change entry point and etc. here...
        return container;
    }
}
```

What do you guys think about it? If you guys think it would be a nice add-on I'll submit a PR for it.

**Comment by eddumelendez** [comment]
Hi @bsofiato, thanks for raising this issue. We would like to understand the use cases for this first. Then we can discuss if we should proceed with a PR or not.

**Comment by kiview** [comment]
As a general note, I feel advanced use cases that require detailed configuration of the container are much better suited to using the existing OOP Testcontainers API, rather than improving the complexity of the JDBC integration (which is already a more specific use case).

**Comment by bsofiato** [comment]
Sure, @eddumelendez. In one project, we used the WNameless Oracle image with OracleAQ.  To make it run correctly, we had to disable Oracle's "asynch.io" feature, which is done by setting an environment variable of the docker image.

@kiview, I feel bootstrapping database containers thru JDBC url is way simpler than providing a custom database. Especially when using SpringBoot as all the container's lifecycle is (indirectly) handled by the spring test framework itself (which is complex since spring might cache test contexts for latter use). 

I might be wrong, but it is up to the developer to manually bootstrap the datasource when using the OOP container approach (which might not be the approach used in production).

**Comment by kiview** [comment]
@bsofiato Are you aware of the [`DynamicPropertySource`](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/test/context/DynamicPropertySource.html) configuration approach in Spring-Boot for setting the respective properties using the container object? It is the recommended way by Spring-Boot and the mechanism was specifically introduced to allow for an easier integration with Testcontainers. It works specifically well with Spring-Context caching and gives you rich controls of instrumenting and bootstrapping the container to your liking.

**Comment by eddumelendez** [comment]
I totally agree with @kiview here. Testcontainers JDBC integration is a convenient way to use however in case there are most complex cases the recommended way is to use the container class for customization.
